### PR TITLE
add tests for paver tasks

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[EDUCATOR-1620](https://openedx.atlassian.net/browse/EDUCATOR-1620)

This adds tests for paver quality tasks.

### CODE REVIEW

**Comment by muzaffaryousaf** [comment]
@muhammad-ammar Can you fix few violations to get green build ?

File: pavelib/tests/test_quality.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    """
+    For testing the paver violations-counting tasks
+    """
+    def setUp(self):
+        super(TestPaverQualityViolations, self).setUp()
+        self.f = tempfile.NamedTemporaryFile(delete=False)
+        self.f.close()
+        self.addCleanup(os.remove, self.f.name)
+
+    def test_pylint_parser_other_string(self):
```
Comment (muzaffaryousaf) [inline pavelib/tests/test_quality.py]: test name is not readable for me, can we change it something like test_pylint_parser_checks_formatting if that's what it's purpose is ?

**Comment by muhammad-ammar** [comment]
@muzaffaryousaf Fixed

File: pavelib/tests/test_quality.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        num = pavelib.quality._count_pylint_violations(f.name)  # pylint: disable=protected-access
+        self.assertEqual(num, 1)
+
+    def test_pep8_parser(self):
+        """
+        Tests that pep8 parser works as expected.
+        """
+        with open(self.f.name, 'w') as f:
+            f.write("hello\nhithere")
+        num, _violations = pavelib.quality._count_pep8_violations(f.name)  # pylint: disable=protected-access
```
Comment (Qubad786) [inline pavelib/tests/test_quality.py]: if we are not using `_violations`, change it to `__`

File: pavelib/tests/test_quality.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+        self.test_code_dir = os.path.join(os.path.dirname(__file__), 'pavelib_test_code')
+
+        # Mock _count_pep8_violations to return a violation
+        self._mock_count_pep8_violations = MagicMock(
+            return_value=(1, ['abc/envs/common.py:32:2: E225 missing whitespace around operator'])
+        )
+
+        # Mock _count_pylint_violations to return a violation
+        self._mock_count_pylint_violations = MagicMock(return_value=1)
```
Comment (Qubad786) [inline pavelib/tests/test_quality.py]: can we use Mock here? because we only need to return a value.

File: pavelib/tests/test_quality.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+    def setUp(self):
+        super(TestPaverQuality, self).setUp()
+        # this is required so that each test will clean environment
+        paver.tasks.environment = paver.tasks.Environment()
+
+        self.test_code_dir = os.path.join(os.path.dirname(__file__), 'pavelib_test_code')
+
+        # Mock _count_pep8_violations to return a violation
+        self._mock_count_pep8_violations = MagicMock(
```
Comment (Qubad786) [inline pavelib/tests/test_quality.py]: can we use Mock here?

File: pavelib/tests/test_quality.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        },
+        {
+            'options': {'limit': 1}
+        },
+        {
+            'options': {'limit': 2}
+        }
+    )
+    @unpack
+    @patch('pavelib.quality.sh', Mock())
```
Comment (Qubad786) [inline pavelib/tests/test_quality.py]: should we also assert the actual shell command via this?

**Comment by muhammad-ammar** [comment]
@muzaffaryousaf @Qubad786 Feedback Addressed

File: pavelib/tests/test_quality.py [hunk 6]
```diff
@@ -0,0 +1,218 @@
+"""
+Tests for paver quality tasks
+"""
+import os
+import tempfile
+import unittest
+
+import paver.tasks
+from paver.easy import call_task
+from ddt import data, ddt, file_data, unpack
```
Comment (Qubad786) [inline pavelib/tests/test_quality.py]: nit: import order.

**Pull request merged**